src/recursion.js

```diff
@@ -0,0 +1,84 @@
+function iteratedFactorial(n) {
+  if (n < 0) {
+    return undefined;
+  }
+
+  if (n === 0) {
+    return 1;
+  }
+
+  let factor = 1;
+
+  for (let i = 1; i <= n; ++i) {
+    factor *= i;
+  }
+
+  return factor;
+}
+
+function recursedFactorial(n) {
+  if (n < 0) {
+    return undefined;
+  }
+
+  if (n === 0) {
+    return 1;
+  }
+
+  return n * recursedFactorial(n - 1);
+}
+
+function iteratedFibonacci(n) {
+  if (n === 0) {
+    return 0;
+  }
+
+  let a = 0;
+  let b = 1;
+
+  for (let i = 1; i < n; ++i) {
+    [a, b] = [b, a + b];
+  }
+
+  return b;
+}
+
+//TODO redo this function using dynamic programming, memoization is also relevant
```

Comment:
nit: it's not you, it's me

```suggestion
// TODO: redo this function using dynamic programming / memoization
```